x-pack/filebeat/input/cel/file_auth.go

```diff
@@ -0,0 +1,97 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package cel
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+	"sync"
+	"time"
+)
+
+type fileAuthTransport struct {
+	header  string
+	prefix  string
+	path    string
+	refresh time.Duration
+	base    http.RoundTripper
+	clock   func() time.Time
+
+	mu       sync.Mutex
+	value    string
+	loadedAt time.Time
+}
+
+func newFileAuthTransport(cfg *fileAuthConfig, base http.RoundTripper) (*fileAuthTransport, error) {
+	if cfg == nil {
+		return nil, fmt.Errorf("file auth: missing configuration")
+	}
+	if base == nil {
+		base = http.DefaultTransport
+	}
+	transport := &fileAuthTransport{
+		header:  cfg.headerName(),
+		prefix:  cfg.Prefix,
+		path:    cfg.Path,
+		refresh: cfg.refreshInterval(),
+		base:    base,
+		clock:   time.Now,
+	}
+	if err := transport.initialise(); err != nil {
+		return nil, err
+	}
+	return transport, nil
+}
+
+func (t *fileAuthTransport) initialise() error {
+	t.mu.Lock()
+	defer t.mu.Unlock()
+	return t.refreshLocked(t.clock())
+}
+
+func (t *fileAuthTransport) RoundTrip(r *http.Request) (*http.Response, error) {
+	value, err := t.currentValue()
+	if err != nil {
+		return nil, err
+	}
+	// Clone the request so we comply with http.RoundTripper requirements and avoid
+	// mutating caller-visible fields such as Header.
+	req := r.Clone(r.Context())
+	if req.Header == nil {
+		req.Header = make(http.Header)
+	} else {
+		req.Header = req.Header.Clone()
+	}
+	req.Header.Set(t.header, value)
+	return t.base.RoundTrip(req)
+}
+
+func (t *fileAuthTransport) currentValue() (string, error) {
+	now := t.clock()
+	t.mu.Lock()
+	defer t.mu.Unlock()
+	if t.value == "" || t.refresh <= 0 || now.Sub(t.loadedAt) >= t.refresh {
+		if err := t.refreshLocked(now); err != nil {
+			return "", err
+		}
+	}
+	return t.value, nil
+}
+
+func (t *fileAuthTransport) refreshLocked(now time.Time) error {
```

Comment:
```suggestion
func (t *fileAuthTransport) refresh(now time.Time) error {
```
The name does no indicate that locked means "must be locked" or "locks, so you don't need to". Let's just use the convention that if something does not state that it's safe for concurrent use, it's not.